crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -338,19 +339,32 @@ impl RecursorDnsHandle {
             }
         }
 
-        // collect missing IP addresses, select over them all, get the addresses
-        // make it configurable to query for all records?
+        // If we have no glue, collect missing IP addresses for non-child NS servers
+        // Querying for child NS servers can result in infinite recursion if the
+        // parent nameserver never returns glue records for child NS records, or does
+        // so based on cache freshness (observed with BIND)
+        //
+        // TODO: make it configurable to query for all records?
```

Comment:
can you describe this in a little more detail? I think you mean use an `ANY` request, but I think `ANY` is being deprecated by many resolvers: https://blog.cloudflare.com/what-happened-next-the-deprecation-of-any

Reply:
I assume you mean the 'TODO' comment - that was already there. I didn't write it.  My understanding of it is that the current logic essentially builds a partial NS lookup pool.  If glue for at least one record is returned by the parent NS, we don't attempt to resolve any other NS records that were included in the delegation response.  I believe the idea here would be to have a config option like 'resolve_all_ns_delegations' in the recursor config to attempt to build a config group based on all provided NS records.

Reply:
ah, could you just remove the TODO? I don't think it's valid any more. Thanks for that clarification, I missed that it was already there.

Reply:
But, in any case, since I didn't resolve it, I left the comment in place.

Reply:
Sure, no problem

Reply:
thank you.